src/bsoncxx/types.hpp

```diff
@@ -122,10 +158,14 @@ struct BSONCXX_API b_utf8 {
     template <typename T,
               typename std::enable_if<!std::is_same<b_utf8, typename std::decay<T>::type>::value,
                                       int>::type = 0>
-    BSONCXX_INLINE explicit b_utf8(T&& t) : value(std::forward<T>(t)) {}
+    BSONCXX_INLINE BSONCXX_DEPRECATED explicit b_utf8(T&& t) : value(std::forward<T>(t)) {}
```

Comment:
We've been able to deprecate a class before gracefully, and I think that the same flow would work for a struct.  Here's what we did for the ssl -> tls rename:

0. Implement new/newly-named NewThing
1. Remove old class Thing
2. Add a deprecated typedef alias for Thing

```
///
/// This class has been renamed to NewThing.
///
/// @deprecated use the NewThing class instead.
///
MONGOCXX_DEPRECATED typedef NewThing Thing;
```

3. Change all instances of Thing -> NewThing in parameters and return values in header files.  Since the alias is there, this shouldn't break user code.

Are there any places that `b_utf8` appears in public headers that you don't think we can get rid of with this strategy?

Reply:
That worked! `b_utf8` is now removed.